search/utils.py

```diff
@@ -33,104 +33,132 @@ def index_indicator(self, indicator):
-            print(indicator, "Error")   # Todo write to error log
-
-    def delete_indicator(self, id):
+        for wf1 in indicator.workflowlevel1.all():
+            if wf1.organization is not None and wf1.organization is not None:
+                org_uuid = str(wf1.organization.organization_uuid)
+
+                # aggregate related models
+                data['workflowlevel1'] = list(map(lambda w: w.name, [wf1]))
+                # data['workflowlevel1'] = list(map(lambda w: w.name, indicator.workflowlevel1.all()))
+
+                # index data with elasticsearch
+                try:
+                    self.es.index(index=self.prefix + org_uuid + "_indicators", id=data['id'], doc_type='indicator',
+                             body=data)
+                except RequestError:
+                    print(indicator, "Error")   # Todo write to error log
+            else:
+                continue
+
+    def delete_indicator(self, indicator):
         if self.es is None:
             return
         try:
-            self.es.delete(index=self.prefix + "indicators", id=id, doc_type='indicator')
+            for wf1 in indicator.workflowlevel1.all():
+                if wf1.organization is not None and wf1.organization is not None:
+                    org_uuid = str(wf1.organization.organization_uuid)
+                    self.es.delete(index=self.prefix + org_uuid + "_indicators", id=indicator.id, doc_type='indicator')
         except NotFoundError:
             logger.warning('Indicator not found in Elasticsearch', exc_info=True)
             raise ValueNotFoundError
 
     def index_workflowlevel1(self, wf):
         if self.es is None:
             return
-        data = self.get_field_data(wf)
 
-        # aggregate related models
-        data = self.get_field_data(wf)
+        if wf.organization is not None:
+            org_uuid = str(wf.organization.organization_uuid)
 
-        # aggregate related models
-        data['sectors'] = list(map(lambda s: s.sector, wf.sector.all()))
-        data['country'] = list(map(lambda c: self.get_field_data(c), wf.country.all()))
+            # aggregate related models
+            data = self.get_field_data(wf)
 
-        # index data with elasticsearch
-        try:
-            self.es.index(index=self.prefix+"workflow_level1", id=data['level1_uuid'], doc_type='workflow', body=data)
-        except RequestError:
-            print(wf, "Error")  # Todo write to error log
+            # aggregate related models
+            data['sectors'] = list(map(lambda s: s.sector, wf.sector.all()))
+            data['country'] = list(map(lambda c: self.get_field_data(c), wf.country.all()))
+
+            # index data with elasticsearch
+            try:
+                self.es.index(index=self.prefix + org_uuid + "_workflow_level1", id=data['level1_uuid'], doc_type='workflow', body=data)
+            except RequestError:
+                print(wf, "Error")  # Todo write to error log
```

Comment:
Do write the error log and don´t print

Reply:
You're right. I overlooked it because it was already there. Will change that now.